packages/medusa/src/api/routes/admin/stock-locations/utils/join-sales-channels.ts

```diff
@@ -17,9 +17,14 @@ const joinSalesChannels = async (
       const salesChannelIds = await channelLocationService.listSalesChannelIds(
         location.id
       )
-      const [salesChannels] = await salesChannelService.listAndCount({
-        id: salesChannelIds,
-      })
+      const [salesChannels] = await salesChannelService.listAndCount(
+        {
+          id: salesChannelIds,
+        },
+        {
+          take: salesChannelIds.length,
+        }
+      )
```

Comment:
question: is this necessary because `take` would otherwise default to 10?

Reply:
The default of 10 could cause issues of there are more than 10 sales channels attached.

I included this as it could pose the same issue as seen with `joinLevels` where not all sales channels would be joined but only the 10 first.